include-build/roborazzi-gradle-plugin/src/test/java/io/github/takahirom/roborazzi/WebAssetsTest.kt

```diff
@@ -0,0 +1,31 @@
+package io.github.takahirom.roborazzi
+
+import junit.framework.TestCase.assertTrue
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import java.io.File
+import kotlin.test.Test
+
+class WebAssetsTest {
+
+  @get:Rule
+  val tmpDir: TemporaryFolder = TemporaryFolder.builder().assureDeletion().build()
+
+  @Test
+  fun testWritesWebAssetsSuccessfullyToReportsDirectory() {
+    val reportDir = tmpDir.newFolder("roborazzi-reports")
+    val cssFile = File(reportDir, "assets/report-style.css")
+    val materializeCssFile = File(reportDir, "assets/materialize.min.css")
```

Comment:
Great!
Could we have a shared file list in WebAssets? I would like to make it easier to update the assets library.

```
val files = assetsPaths.entiries.flatMap{ it.values.map { it.substringAfterLast("/") } }
```

https://github.com/takahirom/roborazzi/blob/ffa622267de2bdba81d1a2222f7f8f7cee5db9e8/include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/WebAssets.kt#L27

```
companion object {
  val assetsPaths = mapOf(
      materializeCss to listOf(
        "css/materialize.min.css",
        "js/materialize.min.js",
      ),
      materialIcons to listOf(
        "material-icons.css",
        "MaterialIcons-Regular.ttf",
      )
    )
}
```

Reply:
Yes we can do that. You want assetsPaths exposed publicly too?

Reply:
Can we use internal or @ VisibleForTesting? But I'm not sure if we can use @ VisibleForTesting in JVM. 👀

Reply:
Oh if it's for use with test, I prefer to keep it private then reconstruct it by hand in test since the values are hardcoded, it can be caught in unit test in case someone changes it in the SUT.

Reply:
I believe we can use `@VisibleForTesting` in jvm tests

Reply:
> Oh if it's for use with test, I prefer to keep it private then reconstruct it by hand in test since the values are hardcoded, it can be caught in unit test in case someone changes it in the SUT.

@takahirom I was hoping to hear your thoughts on this. Should I make it `internal` or `private`?

Reply:
@eyedol 
Sorry for being late. I see your point about tests breaking when the file list changes. While it's not necessarily a bad thing, it does make maintenance a bit challenging. Here are three things I'd like to check in the test:

1. The files we want to include should be in the report folder to verify that index.html can use the files through the shared file list in the SUT.
2. The files should not be empty to confirm that we have the files correctly.
3. The report index.html should include the files to ensure that index.html is using the files.

Reply:
Sorry for the late response. I understand where you coming from though I think in this case it's fine to cause a duplicate as it could potentially catch an issue but yeah, I'll make it internal for reuse then.

Reply:
BTW, I changed the visibility to internal with 3e179ab

Reply:
@takahirom if all is good could you merge when you get the chance otherwise happy to fix anything I missed. Thanks

Reply:
Sorry for being late. What I wanted is to automate this using WebAssets.assetPathsMap so that we don't have to modify tests whenever files are changed.


```
    assertTrue(cssFile.exists())
    assertTrue(materializeCssFile.exists())
    assertTrue(materializeJsFile.exists())
    assertTrue(materialIconsCssFile.exists())
    assertTrue(materialIconsTtfFile.exists())
```

↓

```
val assetFiles = assetsPaths.entiries.flatMap{ it.values }.map { File(it) }

WebAssets.create().writeToRoborazziReportsDir(reportDir)

assetFiles.forEach { file->
  assertTrue(file.exists())
}
```

Reply:
Done with 04b0051